features/player_logistic_requests.lua

```diff
@@ -125,24 +143,34 @@ local player_joined_or_created =
 local research_finished =
     Token.register(
     function(event)
+        -- Invalid tech
         local tech = event.research
-        if not tech or tech.name ~= 'logistic-robotics' then
+        if not (tech and tech.valid) then
             return
         end
 
-        for _, player in pairs(game.connected_players) do
+        -- Already init. or logistics not available
+        local force = tech.force
+        if force and (initialized_forces[force.name] or not force.character_logistic_requests) then
```

Comment:
Shouldn't this be?

```suggestion
        if not force or initialized_forces[force.name] or not force.character_logistic_requests then
```

Previously if `force` was `nil` you could get a nil ref error below for `for _, player in pairs(force.connected_players) do`

Reply:
Technically, `if force and ..` this will already exit in case of `force = nil`. So the parethesis doesnt get computed (accessing fields `.name` and `.character_logistic_requests`

I guess this would be the compact version of 
`if (force and initialized_forces[force.name]) or (force and not force.character_logistic_requests) then`

It would return before getting to the `for .. in pairs(force.conn...)` loop

I can rewrite/make it more explicit if you prefer

Reply:
Maybe I'm being dumb but:
```lua
if (nil and initialized_forces[force.name]) or (nil and not force.character_logistic_requests) then
    return -- This would be unreachable, which I think is the problem as we should return when force is nil.
end
```

Reply:
My god yes, you're absolutely right... I shouldn't review logics this late after work. Will change as well sorry ;)